### ISSUE TITLE
Flakey Relation Integration tests.

### ISSUE THREAD

**Issue by MiaAltieri** [canonical/mongodb-operator#247]
As of recently CI reports [flakey relation integration tests](https://github.com/canonical/mongodb-operator/actions/runs/6179708184/job/16775133065) - seems to have started occuring after updating CI to 3.1.5.

**Comment by MiaAltieri** [comment]
Issue seems to relation error:
```
unit-mongodb-3: 14:39:37 ERROR juju.worker.uniter resolver loop error: cannot enter scope for unit "mongodb/3" in relation "application:multiple-database-clusters mongodb:database": inconsistent state in EnterScope
```
which is currently an error in relations on [Juju 3.1.5 - update to 3.1.6 when available](https://chat.charmhub.io/charmhub/pl/gc8169mqr3fodqymn8fowxw77y)

**Comment by MiaAltieri** [comment]
fixed in #272 and #263
